### ISSUE TITLE
[Bug]: Score == -maxFloat is returned during search

### ISSUE THREAD

**Issue by soothing-rain** [milvus-io/milvus#21138]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:6709ccf690aec88a3323f419b060b8d962b81445
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

It seems a distance == -maxFloat is returned when searching. I doubt if std::numeric_limits<>::lowest() is returned somewhere.

100% reproducible in https://github.com/milvus-io/milvus/issues/21118:
To reproduce, use this repo and uncomment line #762 in https://github.com/ThreadDao/milvus-sdk-go/blob/search/test/testcases/search_test.go . Then go test with `TestSearchInvalidExpr`:
```
go test -race -count=1 -v -run TestSearchInvalidExpr  ./test/testcases
```

Related:
https://github.com/milvus-io/milvus/issues/21118
https://github.com/milvus-io/milvus/issues/20988
https://github.com/milvus-io/milvus/issues/20992

```
[
   {
      "num_queries":5,
      "top_k":10,
      "fields_data":[
         {
            "type":5,
            "Field":{
               "Scalars":{
                  "Data":{
                     "LongData":{
                        "data":[
                           2928,
                           2169,
                           988,
                           1347,
                           1739
                        ]
                     }
                  }
               }
            },
            "field_id":100
         }
      ],
      "scores":[
         -3.4028235e+38,
         -3.4028235e+38,
         -3.4028235e+38,
         -3.4028235e+38,
         -3.4028235e+38
      ],
      "ids":{
         "IdField":{
            "IntId":{
               "data":[
                  2928,
                  2169,
                  988,
                  1347,
                  1739
               ]
            }
         }
      },
      "topks":[
         1,
         1,
         1,
         1,
         1
      ]
   }
]
```

**Comment by soothing-rain** [comment]
> Consult with the issue creator, "score == -maxFloat" is shown in a middle stage of API querynode/result.go::reduceSearchResults(). We regard this as an issue only if "score == -maxFloat" is shown in the final results.

How is it not an issue when Milvus already panicked on this?

**Comment by cydrain** [comment]
Add some debug code, can print out following debug messages when run this case:
```
index type: HNSW
row count: 1503
nq = 5, k = 10, bitset count = 1502
valid bitset: 1
  === No. 0 ===
    {283, 3.40282e+38} 
  === No. 1 ===
    {356, 3.40282e+38} 
  === No. 2 ===
    {1387, 3.40282e+38} 
  === No. 3 ===
    {961, 3.40282e+38} 
  === No. 4 ===
    {439, 3.40282e+38} 
```

**Comment by cydrain** [comment]
after roll back #546 #552 #553 #559, the result is correct
```
nq = 5, k = 10, bitset count = 1502
valid bitset: 1
  === No. 0 ===
    {1, 21.5548} 
  === No. 1 ===
    {1, 22.3684} 
  === No. 2 ===
    {1, 23.1693} 
  === No. 3 ===
    {1, 21.8233} 
  === No. 4 ===
    {1, 21.7304} 
```

**Comment by liliu-z** [comment]
This is fixed in Knowhere 1.3.6. @soothing-rain
